packages/dds/tree/src/test/tableSchema.spec.ts

```diff
@@ -0,0 +1,393 @@
+				index: 0,
+				rows: [
+					{
+						id: "row-0",
+						cells: {},
+					},
+				],
+			});
+
+			assertEqualTrees(treeView.root, {
+				columns: [],
+				rows: [
+					{
+						id: "row-0",
+						cells: {},
+					},
+				],
+			});
+		});
+
+		it("Insert rows into non-empty list", () => {
+			const { treeView } = createTableTree();
+			treeView.initialize({
+				rows: [
+					{
+						id: "row-a",
+						cells: {},
+					},
+					{
+						id: "row-b",
+						cells: {},
+					},
+				],
+				columns: [],
+			});
+
+			treeView.root.insertRows({
+				index: 1,
+				rows: [
+					{
+						id: "row-c",
+						cells: {},
+					},
+					{
+						id: "row-d",
+						cells: {},
+					},
+				],
+			});
+
+			assertEqualTrees(treeView.root, {
+				columns: [],
+				rows: [
+					{
+						id: "row-a",
+						cells: {},
+					},
+					{
+						id: "row-c",
+						cells: {},
+					},
+					{
+						id: "row-d",
+						cells: {},
+					},
+					{
+						id: "row-b",
+						cells: {},
+					},
+				],
+			});
+		});
+
+		it("Append rows", () => {
+			const { treeView } = createTableTree();
+			treeView.initialize({
+				rows: [
+					{
+						id: "row-a",
+						cells: {},
+					},
+					{
+						id: "row-b",
+						cells: {},
+					},
+				],
+				columns: [],
+			});
+
+			// By not specifying an index, the column should be appended to the end of the list.
+			treeView.root.insertRows({
+				rows: [
+					{
+						id: "row-c",
+						cells: {},
+					},
+					{
+						id: "row-d",
+						cells: {},
+					},
+				],
+			});
+
+			assertEqualTrees(treeView.root, {
+				columns: [],
+				rows: [
+					{
+						id: "row-a",
+						cells: {},
+					},
+					{
+						id: "row-b",
+						cells: {},
+					},
+					{
+						id: "row-c",
+						cells: {},
+					},
+					{
+						id: "row-d",
+						cells: {},
+					},
+				],
+			});
+		});
+
+		// Test TODOs:
+		// - Failure case: insert row(s) that already exist in the tree.
+	});
+
+	describe("Set cell", () => {
+		it("Append rows", () => {
+			const { treeView } = createTableTree();
+			treeView.initialize({
+				columns: [
+					{
+						id: "column-0",
+					},
+				],
+				rows: [
+					{
+						id: "row-0",
+						cells: {},
+					},
+				],
+			});
+
+			// By not specifying an index, the column should be appended to the end of the list.
+			treeView.root.setCell({
+				key: {
+					rowId: "row-0",
+					columnId: "column-0",
+				},
+				cell: { value: "Hello world!" },
+			});
+
+			assertEqualTrees(treeView.root, {
+				columns: [
+					{
+						id: "column-0",
+					},
+				],
+				rows: [
+					{
+						id: "row-0",
+						cells: {
+							"column-0": {
+								value: "Hello world!",
+							},
+						},
+					},
+				],
+			});
+		});
+
+		// Test TODOs:
+		// - Failure case: insert cell that already exist in the tree.
+		// - Failure case: Insert cell with invalid row/column id.
+	});
+
+	describe("Remove column", () => {
+		// Test TODOs:
+		// - Success case: remove existing column.
```

Comment:
I ran out of time to get more tests done. @daesun-park if you have time over the next couple weeks, these would be fair game 😁. Otherwise, I can tackle them when I get back.